src/GafferScene/UDIMQuery.cpp

```diff
@@ -0,0 +1,315 @@
+}
+
+void UDIMQuery::affects( const Plug *input, AffectedPlugsContainer &outputs ) const
+{
+	ComputeNode::affects( input, outputs );
+
+	if(
+		input == uvNamePlug() ||
+		input == extraAttributesPlug() ||
+		input == filterPlug() ||
+		input == scenePlug()->objectPlug() ||
+		input == scenePlug()->attributesPlug()
+	)
+	{
+		outputs.push_back( udimInformationPlug() );
+	}
+}
+
+namespace
+{
+
+struct InfoHashAccumulator
+{
+	bool operator()( const GafferScene::ScenePlug *scene, const GafferScene::ScenePlug::ScenePath &path )
+	{
+		IECore::MurmurHash locationHash;
+		scene->objectPlug()->hash( locationHash );
+		scene->attributesPlug()->hash( locationHash );
+		locationHash.append( scene->objectHash( path ) );
+		std::string pathStr;
+		ScenePlug::pathToString( path, pathStr );
+		m_hashes.push_back( std::pair<std::string, IECore::MurmurHash>( pathStr, locationHash ) );
+		return true;
+	}
+
+	void appendHash( IECore::MurmurHash &h )
+	{
+		std::sort( m_hashes.begin(), m_hashes.end() );
+		for( const auto &i : m_hashes )
+		{
+			h.append( i.second );
+		}
+	}
+
+	tbb::concurrent_vector< std::pair< std::string, IECore::MurmurHash > > m_hashes;
+};
+
+} // namespace
+
+void UDIMQuery::hash( const Gaffer::ValuePlug *output, const Gaffer::Context *context, IECore::MurmurHash &h ) const
+{
+	ComputeNode::hash( output, context, h );
+	if( output == udimInformationPlug() )
+	{
+		uvNamePlug()->hash( h );
+		extraAttributesPlug()->hash( h );
+
+		InfoHashAccumulator f;
+		GafferScene::SceneAlgo::filteredParallelTraverse( scenePlug(), filterPlug(), f );
+		f.appendHash( h );
+	}
+}
+
+namespace {
+
+struct BakeInfoData
+{
+	std::string mesh;
+	std::vector<int> udims;
+	ConstCompoundObjectPtr extraAttributes = nullptr;
+};
+
+struct InfoDataAccumulator
+{
+	InfoDataAccumulator( std::string uvName, std::string extraAttributes )
+		: m_uvName( uvName ), m_extraAttributes( extraAttributes )
+	{
+	}
+
+	bool operator()( const GafferScene::ScenePlug *scene, const GafferScene::ScenePlug::ScenePath &path )
+	{
+		BakeInfoData info;
+		ScenePlug::pathToString( path, info.mesh );
+		IECore::CompoundObjectPtr locationDict;
+
+		IECore::ConstObjectPtr object = scene->objectPlug()->getValue();
+		const IECoreScene::MeshPrimitive *meshPrimitive = runTimeCast<const IECoreScene::MeshPrimitive>( object.get() );
+		if( !meshPrimitive ) return true;
+
+	
+		auto uvs = meshPrimitive->variableIndexedView<IECore::V2fVectorData>( m_uvName,  IECoreScene::PrimitiveVariable::FaceVarying );
+		if( !uvs ) return true;
+
+		std::set<int> udims;
+
+		const IntVectorData *vertsPerFaceData = meshPrimitive->verticesPerFace();
+		const std::vector<int> &vertsPerFace = vertsPerFaceData->readable();
+
+		int faceVertId = 0;	
+		for( int numVerts : vertsPerFace )
+		{
+			Imath::V2f accum = Imath::V2f(0);
+			for( int i = 0; i < numVerts; i++ )
+			{
+				accum += (*uvs)[faceVertId];
+				faceVertId++;
+			}
+			Imath::V2f centerUV = accum / numVerts;
```

Comment:
I'm curious about this. Is it because if we treat each vertex individually, we can end up with rounding taking us over into the next tile? Might be worth a comment explaining why we don't treat vertices individually.

Reply:
Ping.